lib/plugins/game.js

```diff
@@ -24,8 +24,10 @@ function inject (bot, options) {
 
   function handleRespawnPacketData (packet) {
     bot.game.levelType = packet.levelType ?? (packet.isFlat ? 'flat' : 'default')
-    bot.game.hardcore = packet.isHardcore ?? Boolean(packet.gameMode & 0b100)
-    bot.game.gameMode = parseGameMode(packet.gameMode)
+    if (packet.gameMode != undefined) {
```

Comment:
This should check either for `packet.gameMode != null` or use !== (strict equality).